junit/src/main/java/org/kaazing/k3po/junit/rules/SpecificationStatement.java

```diff
@@ -112,10 +122,11 @@ public void evaluate() throws Throwable {
                 }
             }
 
+            // Stefan - not sure what the comment bellow is about. There is no join() method in the statement or rule class.
```

Comment:
The method `apply()` in the rules are called sequential, each one with the statement resulted from calling the previous rule. All this is in one thread, you cannot get the parameters to call the next rule before the previous one ends. So, I'm not sure what you mean by :

> rules after it (or maybe before depending on your mental ordering) complete before the K3po rule has completed execution

Even in `RuleChain`, the code in `apply()` is :
```
for (TestRule each : rulesStartingWithInnerMost) {
            base = each.apply(base, description);
        }
```
No multi-threading here...

I also think that the comment is about some checks that should be done in this piece of code (I suppose it is done in that assetTrue), not about some unit tests/integration tests.